docs/source/users/problem_definition_files/spinw.rst

```diff
@@ -15,105 +32,132 @@ software, name, description
-
-sample
-  The sample defines the target of the experiment and has several parameters.
-
-  This must have a class as "IX_Sample", the remaining arguments are passed
-  through to the IX_Sample class constructor.
-  Full details can be found `here <https://pace-neutrons.github.io/Horace/v3.6.2/user_guide/Resolution_convolution.html#the-tobyfit-class>`__.
+  For SpinW we require an SQW file or a MAT file containing the data from Horace. 
   
-  e.g.::
-    sample = 'class=IX_sample,xgeom=[0,0,1],ygeom=[0,1,0],shape='cuboid',ps=[0.01,0.05,0.01])
+.. note::
+  The MAT file should be used if you are loading in multiple objects. Make sure 
+  to load the data in appropriately in the `wye_function`. 
+
+function
+  The function is defined by one or two matlab files which returns a SpinW model of the foreground or the foreground and background
+  repectively separated by a semicolon.
+
+  The format is again comma seperated key-value pairs, with the matlab files 
+  defined by the variable "foreground" and "background" and the remaining pairs defining starting
+  values as in the native parser.
+
+  Examples:
+
+    Only foreground: 
+    
+    .. code-block:: rst
+
+      function = 'foreground=m_scripts/functions/mftest_gauss_bkgd.m ,height=100,centre=50,sigma=7,const=0,grad=0'
 
-instrument
-  The instrument defines where the measurements were taken and again requires
-  a ``class`` followed by the correct arguments for the class.
+    Foreground and background:
+
+    .. code-block:: rst
+
+      function = 'foreground=m_scripts/functions/gauss.m ,height=1100,centre=66 ,stdev=13; background=m_scripts/functions/linear_bg.m ,bkgd_const=15'
 
-  As with sample, details can be found `here <https://pace-neutrons.github.io/Horace/v3.6.2/user_guide/Resolution_convolution.html#the-tobyfit-class>`__.
   
-  e.g.::
-    instrument = 'class=maps_instrument,ei=70,hz=250,chopper=S'
+wye_function
+  The wye_function is defined by a matlab file which returns the `w` (this could be a sqw ,d1d, d2d, d3d and d4d object), 
+  `e` (this is the standard deviation arrays from the objects), `y` (intensity arrays from the objects), along with a  
+  mask array (`msk`) that indicates which elements are to be retained (where elements of msk are true, the corresponding elements of 
+  y and e are retained). This matlab file takes in the path of the datafile and the path of where the matlab functions are located.
 
-function
-  The function is defined by a matlab file which returns a SpinW model.
+  Explained example of the wye_function:
 
-  The format is again comma seperated key-value pairs, with the matlab file
-  defined by the variable "matlab_script" and the remaining pairs defining starting
-  values as in the native parser.
+  The first three lines adds the path to matlab functions need for fitting and loads the `w` object.
 
-  e.g.::
-    function = 'matlab_script=pcsmo_model.m,J=35,D=0,gam=30,temp=10,amp=300'
+  .. code-block:: rst
```

Comment:
in reST the `rst` here (I believe) should be the language of the following code (i.e. matlab if a role has been defined). @AndrewLister-STFC may know better.

Reply:
Yes, something like:

```
.. codeblock:: python
```

In this case I guess python is the closest but I doubt syntax highlighting will give any insight so `rst` is probably fine.